### PR TITLE
Actuators3

### LINKED ISSUE #341: Add Another Extension to Use Most/All Actuators
Provide functionality similar to `AddCloudFoundryActuators` that isn't CF-specific

### PR DESCRIPTION
**Pull request by hananiel**
resolves #341 #253 

* Removes Obsoleted Code
* Removes newtonSoft in Management (to System.Text.Json)
* Makes middleware endpoint enabled - Which allows you to Map<TEndpoint>
* Added an AddAllActuators HostBuilder extension
* Added code coverage for HealthRegistrations

### CODE REVIEW

**Comment by TimHess** [comment]
With the removal of most of the `*ApplicationBuilderExtensions`, are you expecting that people will no longer use `startup.cs` to add actuators, or that they'll just call `endpoints.Map<TEndpoint>();` if/when they don't want to use the `*HostBuilderExtensions`?

File: src/Management/test/CloudFoundryCore.Test/CloudFoundryHostBuilderExtensionsTest.cs [hunk 1]
```diff
@@ -124,11 +126,28 @@ public async Task AddCloudFoundryActuators_IHostBuilder_IStartupFilterFires()
...
-                .ConfigureWebHost(c => c.UseTestServer().Configure(app => { }))
+                .ConfigureWebHost(testServerWithRouting)
                 .ConfigureAppConfiguration(cbuilder => cbuilder.AddInMemoryCollection(managementSettings));
 
             // Act
-            var host = await hostBuilder.AddCloudFoundryActuators(MediaTypeVersion.V1, ActuatorContext.ActuatorAndCloudFoundry).StartAsync();
+            var host = await hostBuilder.AddCloudFoundryActuators(MediaTypeVersion.V2).StartAsync();
+
+            // Assert general success...
+            //   not sure how to actually validate the StartupFilter worked,
```
Comment (TimHess) [inline src/Management/test/CloudFoundryCore.Test/CloudFoundryHostBuilderExtensionsTest.cs:136]: Check [via HTTP](https://docs.microsoft.com/en-us/aspnet/core/test/middleware?view=aspnetcore-3.1)

**Comment by hananiel** [comment]
> expecting that people will no longer use `startup.cs` to add actuators, or that they'll just call `endpoints.Map<TEndpoint>();` if/when they don't want to use the `*Ho`

They will have to use Map<TEndpoint>

File: src/Management/test/EndpointCore.Test/ActuatorRouteBuilderExtensionsTest.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+using Xunit;
+
+namespace Steeltoe.Management.Endpoint
+{
+    public class ActuatorRouteBuilderExtensionsTest
+    {
+        public static IEnumerable<object[]> IEndpointImplementations
+        {
+            get
+            {
```
Comment (TimHess) [inline src/Management/test/EndpointCore.Test/ActuatorRouteBuilderExtensionsTest.cs:18]: This is crafty, I like it

File: src/Management/test/EndpointCore.Test/Refresh/EndpointMiddlewareTest.cs [hunk 3]
```diff
@@ -48,7 +50,7 @@ public async void HandleRefreshRequestAsync_ReturnsExpected()
             context.Response.Body.Seek(0, SeekOrigin.Begin);
             var reader = new StreamReader(context.Response.Body, Encoding.UTF8);
             var json = await reader.ReadLineAsync();
-            var expected = "[\"management\",\"management:endpoints\",\"management:endpoints:path\",\"management:endpoints:enabled\",\"Logging\",\"Logging:LogLevel\",\"Logging:LogLevel:Steeltoe\",\"Logging:LogLevel:Pivotal\",\"Logging:LogLevel:Default\",\"Logging:IncludeScopes\"]";
+            var expected = "[\"management\",\"management:endpoints\",\"management:endpoints:enabled\",\"Logging\",\"Logging:LogLevel\",\"Logging:LogLevel:Steeltoe\",\"Logging:LogLevel:Pivotal\",\"Logging:LogLevel:Default\",\"Logging:IncludeScopes\"]";
```
Comment (TimHess) [inline src/Management/test/EndpointCore.Test/Refresh/EndpointMiddlewareTest.cs:53]: Not necessary in this PR, but we can/should remove references to Pivotal namespaces, I don't think they're relevant anymore

File: src/Management/src/EndpointCore/ActuatorRouteBuilderExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+                var fullPath = options.GetContextPath(mgmtOptions);
+
+                var pipeline = endpoints.CreateApplicationBuilder()
+                    .UseMiddleware(middleware, mgmtOptions)
+                    .Build();
+
+                if (options.AllowedVerbs == null)
+                {
+                    endpoints.Map(fullPath, pipeline);
```
Comment (TimHess) [inline src/Management/src/EndpointCore/ActuatorRouteBuilderExtensions.cs:102]: Will this result in all verbs being mapped? I'd think we'd want to limit it to `GET` by default

File: src/Management/src/EndpointCore/ActuatorServiceCollectionExtensions.cs [hunk 5]
```diff
@@ -11,10 +25,42 @@ public static class ActuatorServiceCollectionExtensions
...
+            }
+
+            if (config == null)
+            {
+                throw new ArgumentNullException(nameof(config));
+            }
+
+            services.AddHypermediaActuator(config);
+
+            if (Environment.OSVersion.Platform == PlatformID.Win32NT)
```
Comment (TimHess) [inline src/Management/src/EndpointCore/ActuatorServiceCollectionExtensions.cs:48]: We should be adding Heap dump on linux too... #100

File: src/Stream/src/Base/Binder/AbstractMessageChannelBinder.cs [hunk 6]
```diff
@@ -555,20 +555,7 @@ private void EnhanceMessageChannel(IMessageChannel inputChannel)
...
-                ErrorInfrastructure = errorInfrastructure;
-            }
-
-            protected internal IMessageSource Source { get; }
-
-            protected internal ErrorInfrastructure ErrorInfrastructure { get; }
-        }
-
-        protected class ErrorInfrastructure
+        public class ErrorInfrastructure
```
[the remark this answers is not in the document]
Comment (hananiel) [reply src/Stream/src/Base/Binder/AbstractMessageChannelBinder.cs:558]: This was a warning I got so I fixed it - not sure how that got merged in.

File: src/Management/src/EndpointCore/SpringBootAdminClient/HttpClientExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```
[the remark this answers is not in the document]
Comment (hananiel) [reply src/Management/src/EndpointCore/SpringBootAdminClient/HttpClientExtensions.cs:1]: Great - didn't know about it but now its there.

File: src/Management/test/CloudFoundryCore.Test/CloudFoundryHostBuilderExtensionsTest.cs (hunk 1 above)
Comment (hananiel) [reply src/Management/test/CloudFoundryCore.Test/CloudFoundryHostBuilderExtensionsTest.cs:136]: This was your comment I believe - from prior versions - should I remove the comment ? we have other places where we make a call?

File: src/Management/src/EndpointCore/ActuatorRouteBuilderExtensions.cs (hunk 4 above)
Comment (hananiel) [reply src/Management/src/EndpointCore/ActuatorRouteBuilderExtensions.cs:85]: You are correct - good catch!

File: src/Management/src/EndpointCore/ActuatorServiceCollectionExtensions.cs (hunk 5 above)
Comment (hananiel) [reply src/Management/src/EndpointCore/ActuatorServiceCollectionExtensions.cs:48]: Added

File: src/Management/test/CloudFoundryCore.Test/CloudFoundryHostBuilderExtensionsTest.cs (hunk 1 above)
Comment (TimHess) [reply src/Management/test/CloudFoundryCore.Test/CloudFoundryHostBuilderExtensionsTest.cs:136]: hah, that figures... we should either remove the comments or make the http calls. I know there's more than one instance of this comment but I don't know offhand if each startupfilter has a test that calls.... I think there should be at least one http call for each

File: src/Management/src/EndpointCore/ActuatorRouteBuilderExtensions.cs [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+                }
+                else
+                {
+                    endpoints.Map<ThreadDumpEndpoint>();
+                }
+
+                endpoints.Map<HeapDumpEndpoint>();
+            }
+
+            if (Environment.OSVersion.Platform == PlatformID.Win32NT || Platform.IsLinux)
```
Comment (TimHess) [inline src/Management/src/EndpointCore/ActuatorRouteBuilderExtensions.cs:107]:
```suggestion
            if (Platform.IsWindows || Platform.IsLinux)
```

File: src/Management/src/EndpointCore/ActuatorRouteBuilderExtensions.cs [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+                var allowedVerbs = options.AllowedVerbs ?? new List<string> { "Get" };
+
+                endpoints.MapMethods(fullPath, allowedVerbs, pipeline);
+            }
+        }
+
+        public static void MapAllActuators(this IEndpointRouteBuilder endpoints, MediaTypeVersion version = MediaTypeVersion.V2)
+        {
+            endpoints.Map<ActuatorEndpoint>();
+            if (Environment.OSVersion.Platform == PlatformID.Win32NT)
```
Comment (TimHess) [inline src/Management/src/EndpointCore/ActuatorRouteBuilderExtensions.cs:93]:
```suggestion
            if (Platform.IsWindows)
```

File: src/Management/src/EndpointCore/ActuatorServiceCollectionExtensions.cs [hunk 10]
```diff
@@ -11,10 +27,46 @@ public static class ActuatorServiceCollectionExtensions
...
```
Comment (TimHess) [inline src/Management/src/EndpointCore/ActuatorServiceCollectionExtensions.cs:50]:
```suggestion
            if (Platform.IsWindows)
```

File: src/Management/src/EndpointCore/ActuatorRouteBuilderExtensions.cs (hunk 9 above)
Comment (hananiel) [reply src/Management/src/EndpointCore/ActuatorRouteBuilderExtensions.cs:93]: refactored to use Platform.IsWindows everywhere

File: src/Management/src/EndpointCore/ActuatorServiceCollectionExtensions.cs (hunk 10 above)
Comment (hananiel) [reply src/Management/src/EndpointCore/ActuatorServiceCollectionExtensions.cs:50]: refactored to use Platform.IsWindows everywhere

File: src/Management/src/EndpointCore/ActuatorRouteBuilderExtensions.cs [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+                if (version == MediaTypeVersion.V2)
+                {
+                    endpoints.Map<ThreadDumpEndpoint_v2>();
+                }
+                else
+                {
+                    endpoints.Map<ThreadDumpEndpoint>();
+                }
+
+                endpoints.Map<HeapDumpEndpoint>();
```
Comment (TimHess) [inline src/Management/src/EndpointCore/ActuatorRouteBuilderExtensions.cs:104]:
```suggestion
```

repeated below

File: src/Management/src/EndpointCore/ActuatorServiceCollectionExtensions.cs [hunk 12]
```diff
@@ -11,10 +27,45 @@ public static class ActuatorServiceCollectionExtensions
...
             var mgmtOptions = services.BuildServiceProvider().GetServices<IManagementOptions>();
-            foreach (var mgmt in mgmtOptions)
+            foreach (var mgmtOption in mgmtOptions)
             {
-                mgmt.EndpointOptions.Add(options);
+                mgmtOption.EndpointOptions.Add(options);
             }
         }
+
+        public static void AddAllActuators(this IServiceCollection services, IConfiguration config, MediaTypeVersion version)
```
Comment (TimHess) [inline src/Management/src/EndpointCore/ActuatorServiceCollectionExtensions.cs:36]:
```suggestion
        public static void AddAllActuators(this IServiceCollection services, IConfiguration config, MediaTypeVersion version = MediaTypeVersion.V2)
```

**Pull request merged**